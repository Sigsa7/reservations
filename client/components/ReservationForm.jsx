import React from 'react';
import PropTypes from 'prop-types';

const ReservationForm = ({ restaurant }) => (
  <div>
    {restaurant}
  </div>
);

ReservationForm.propTypes = {
  restaurant: PropTypes.shape({
    open_time: PropTypes.shape({
      hour: PropTypes.number,
      minute: PropTypes.number,
    }),
    close_time: PropTypes.shape({
      hour: PropTypes.number,
      minute: PropTypes.number,
    }),
    time_intervals: PropTypes.number,
    max_seating: PropTypes.number,
    max_party_size: PropTypes.number,
  }).isRequired,
};

export default ReservationForm;
