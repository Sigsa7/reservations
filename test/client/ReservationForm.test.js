import React from 'react';
import renderer from 'react-test-renderer';

import ReservationForm from '../../client/components/ReservationForm';

describe('ReservationForm component', () => {
  const sampleRestaurant = {
    open_time: {
      hour: 0,
      minute: 0,
    },
    close_time: {
      hour: 0,
      minute: 0,
    },
    time_intervals: 0,
    max_seating: 0,
    max_party_size: 0,
  };

  it('renders with no errors when valid restaurant passed in', () => {
    const component = renderer.create(<ReservationForm restaurant={sampleRestaurant} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
