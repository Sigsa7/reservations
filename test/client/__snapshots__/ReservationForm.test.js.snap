// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationForm component renders with no errors when valid restaurant passed in 1`] = `
<div>
  {"open_time":{"hour":0,"minute":0},"close_time":{"hour":0,"minute":0},"time_intervals":0,"max_seating":0,"max_party_size":0}
</div>
`;
